apps/include/apps.h

```diff
@@ -132,9 +126,14 @@ __owur int ctx_set_verify_locations(SSL_CTX *ctx,
  */
 __owur int ctx_set_ctlog_list_file(SSL_CTX *ctx, const char *path);
 
-#endif
+# endif
 
-ENGINE *setup_engine(const char *engine, int debug);
+ENGINE *setup_engine_flags(const char *engine, unsigned int flags, int debug);
+# ifndef OPENSSL_NO_ENGINE
+#  define setup_engine(e, debug) setup_engine_flags(e, ENGINE_METHOD_ALL, debug)
+# else
+#  define setup_engine(e, debug) setup_engine_flags(e, 0, debug)
+# endif
```

Comment:
1.  "flags" is a misnomer.  "methods" would be more accurate.  ENGINE *flags* is a whole different set of macros.
2.  There's really no need for two variants of this macro.  Since `ENGINE_METHOD_ALL` is literally all possible bits set and no one checks outside of the currently known range, you can have this:

    ``` C
    # define setup_engine(e, debug) setup_engine_flags(e, (unsigned int)-1, debug)
    ```

    That should be perfectly safe, and acceptable within the app.

Reply:
Thanks @levitte for your comments! Done.